streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -3062,7 +3062,7 @@ <GK, GV, RV> KStream<K, RV> leftJoin(final GlobalKTable<GK, GV> globalTable,
      * @see #map(KeyValueMapper)
      */
     <KOut, VOut> KStream<KOut, VOut> process(
-        final ProcessorSupplier<? super K, ? super V, KOut, VOut> processorSupplier,
+        final ProcessorSupplier<? super K, ? super V, ? extends KOut, ? extends VOut> processorSupplier,
```

Comment:
While this is technically public API, to me it's a bug-fix, so would not want to do a KIP for it... Thoughts?

Reply:
Yeah, this should be covered by KIP-100 in spirit, I don't think you need a KIP.

Reply:
@mjsax sorry I just saw this from the community slack --- I read the JIRA ticket and the PR itself, and this looks great to me (we should have done it a while ago, thanks for completing it). As for https://github.com/apache/kafka/pull/18721#discussion_r1934537661 I gave some thought on it, and KLeft/KRight/VLeft/VRight sounds reasonable as well.